### PR TITLE
SAO work: ActiveObject types & SAO cleanup

### PR DESCRIPTION
**Pull request by nerzhul**
* Replace u8 types with ActiveObjectType.
* Merge content_object.h into activeobject.h
* Remove DummyLoadSAO, it's now unused.

This is a cleanup for deadcode and use an enum for better typing ActiveObjects

### CODE REVIEW

**Comment by Novatux** [comment]
Looks good

**Comment by nerzhul** [comment]
It will be merged in ~12hours then

File: src/activeobject.h [hunk 1]
```diff
@@ -23,7 +23,16 @@ with this program; if not, write to the Free Software Foundation, Inc.,
 #include "irr_aabb3d.h"
 #include <string>
 
-#define ACTIVEOBJECT_TYPE_INVALID 0
+enum ActiveObjectType {
+	ACTIVEOBJECT_TYPE_INVALID = 0,
+	ACTIVEOBJECT_TYPE_TEST = 1,
+	ACTIVEOBJECT_TYPE_ITEM = 2,
```
Comment (ShadowNinja) [inline src/activeobject.h]: Enums start at 0 and increase by one from the last element, so these don't need to be specified explicitly.  (but meh, minor)

**Comment by ShadowNinja** [comment]
:+1:

File: src/activeobject.h (hunk 1 above)
Comment (nerzhul) [inline src/activeobject.h]: In fact they are very old, it's 0.3 related. Discuting with mod maintainers and nore it can be safely removed.

**Comment by nerzhul** [comment]
Commit f8d5af753617d502920556cff88f451ef670c210 (which also remove unused ItemSAO)

**Pull request closed without merging**